src/inmanta/agent/io/remote.py

```diff
@@ -125,10 +129,19 @@ def _build_connect_string(self) -> str:
 
     def _execute(self, function_name, *args, **kwargs):
         with self._lock:
-            ch = self._gw.remote_exec(local)
-            ch.send((function_name, args, kwargs))
-            result = ch.receive()
-            ch.close()
+            try:
+                ch = self._gw.remote_exec(local)
+            except OSError:
+                raise ChannelClosedException(
+                    "Failed to execute remote command on %(uri)s, this is most likely caused by "
+                    "either the target machine being unavailable "
+                    "or the python command being unavailable on the target machine.",
+                    remote_host=self._host + ":" + str(self._port)
```

Comment:
Why is this remote_host argument here?

Reply:
because the uri actually isn't accessible here, and I forgot to replace it higher up... I replaced uri with an f-string and populated it with the host and port.